corehq/apps/sms/models.py

```diff
@@ -1078,6 +1079,12 @@ class MessagingEvent(models.Model, MessagingStatusMixin):
     class Meta(object):
         app_label = 'sms'
 
+    def get_source_display(self):
+        # for some reason source choices aren't set in the field, so manually add this method.
```

Comment:
this was removed for reasons that aren't totally clear in https://github.com/dimagi/commcare-hq/commit/6bdcf20a30a9dc6241e383fd48983316c25a0d87. I didn't want to add it back to limit the scope of this change.

Reply:
Yeah, odd - I wonder whether it was just so subsequent source options could be added without triggering a migration